File: #4. 2021 카카오 채용연계형 인턴십/장효빈/problem1_숫자_문자열과_영단어.py
# 풀이 시간: 10분
# 시간 복잡도: O(N * 10)
# 코드 설명: 매핑해놓고 교체
# 취약한 부분: 

def solution(s):
    d = {'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 
         'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}
    
    for num in d.keys():
        s = s.replace(num, d[num])
        
    return int(s)

File: #4. 2021 카카오 채용연계형 인턴십/김민정/problem1_숫자문자열과영단어.java
// 풀이 시간: 잘 기억이 안 납니다...
// 시간 복잡도: O(N)
// 코드 설명: replace로 다 바꿨습니다
// 취약한 부분: ...그냥 구현한 듯..?

class Solution {
    public int solution(String s) {
        int answer = 0;
        
        String tmp = s.replace("zero", "0").replace("one", "1")
                     .replace("two", "2").replace("three", "3")
                     .replace("four", "4").replace("five", "5")
                     .replace("six", "6").replace("seven", "7")
                     .replace("eight", "8").replace("nine", "9");
        
        return Integer.parseInt(tmp);
    }
}
